config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Encass.java

```diff
@@ -156,4 +160,34 @@ public void sortArgumentsAndResults() {
         setArguments(getArguments().stream().collect(toCollection(() -> new TreeSet<>(Comparator.comparing(EncassArgument::getName)))));
         setResults(getResults().stream().collect(toCollection(() -> new TreeSet<>(Comparator.comparing(EncassResult::getName)))));
     }
+
+    @Override
+    public AnnotatedEntity<GatewayEntity> getAnnotatedEntity(final String projectName,
```

Comment:
**Avoid taking project name and version**. You might earn something down the line. But, from design, why would I inject project details while getting the annotated entity.

Reply:
yeah, i felt the same, actually we can create annotated entity when loading the entity from yml files as it is design time info